lib/3scale/client.rb

```diff
@@ -122,20 +124,27 @@ def authrep(options)
     #
     # == Examples
     #
-    #   # Report two transactions of two applications.
-    #   client.report({:app_id => 'foo', :usage => {'hits' => 1}},
-    #                 {:app_id => 'bar', :usage => {'hits' => 1}})
+    #    Report two transactions of two applications. Using the default service.
+    #    client.report(transactions: [{:app_id => 'foo', :usage => {'hits' => 1}},
+    #                                 {:app_id => 'bar', :usage => {'hits' => 1}}])
     #
-    #   # Report one transaction with timestamp.
-    #   client.report({:app_id    => 'foo',
-    #                  :timestamp => Time.local(2010, 4, 27, 15, 14),
-    #                  :usage     => {'hits' => 1})
+    #    Report one transaction with timestamp. Using the default service.
+    #    client.report(transactions: [{:app_id    => 'foo',
+    #                                  :timestamp => Time.local(2010, 4, 27, 15, 14),
+    #                                  :usage     => {'hits' => 1}])
     #
-    def report(*transactions)
-      raise ArgumentError, 'no transactions to report' if transactions.empty?
+    #    Report a transaction specifying the service.
+    #    client.report(transactions: [{:app_id => 'foo', :usage => {'hits' => 1}}],
+    #                  service_id: 'a_service_id')
+    #
+    def report(transactions: [], service_id: nil)
```

Comment:
One thing to keep in mind is that `transactions.concat(reports).concat([rest])` may add an empty object to the `transactions` array (if `transactions: []` param is used), which will later break in `encode_transactions`.

So, either it should be filtered out before, or validated first inside `encode_transactions`.